dialogue-target/src/main/java/com/palantir/dialogue/BodySerDe.java

```diff
@@ -41,9 +43,13 @@ public interface BodySerDe {
      */
     Deserializer<InputStream> inputStreamDeserializer();
 
+    <T> Deserializer<T> inputStreamDeserializer(DeserializerArgs<T> deserializerArgs);
```

Comment:
I also added methods to deserialize input streams and optional input streams. The idea is that the type T here has a constructor with `InputStream` as the only parameter, and on a successful binary response body, a `SuccessT(response.body())` will be returned.